app/services/CheckoutService.scala

```diff
@@ -75,7 +75,12 @@ class CheckoutService(
     } yield {
       updateAuthenticatedUserDetails()
       sendETDataExtensionRow(result)
+      subscribe.paymentMethodOpt match {
```

Comment:
Maybe this should go inside the for comprehension, so that the subscribe action fails if this request fails? In membership-frontend that's how it works: https://github.com/guardian/membership-frontend/blob/master/frontend/app/services/MemberService.scala#L118-L119